File: THCS/2023_2024/CanTho_2324/Bai4/BANGSO.cpp
#include <iostream>
#include <fstream>

using namespace std;
void count_occurrences(const string &input_file, const string &output_file) {
    ifstream infile(input_file);
    ofstream outfile(output_file);

    long long n, k;
    infile >> n >> k;

    long long count = 0;

    // Tìm tất cả các ước số của k
    for (long long i = 1; i * i <= k; ++i) {
        if (k % i == 0) { // i là ước của k
            long long j = k / i; // Tính j

            // Kiểm tra (i, j) và (j, i) có hợp lệ hay không
            if (i <= n && j <= n) {
                count++; // Đếm (i, j)
            }
            if (i != j && j <= n && i <= n) {
                count++; // Đếm (j, i) nếu khác nhau
            }
        }
    }

    outfile << count << endl;

    infile.close();
    outfile.close();
}

int main() {
    count_occurrences("BANGSO.INP", "BANGSO.OUT");

    return 0;
}

File: THCS/2023_2024/CanTho_2324/Bai4/BANGSO.py
input_path = "./BANGSO.INP"
output_path = "./BANGSO.OUT"
def count_occurrences(input_file, output_file):
    with open(input_file, 'r') as infile:
        n, k = map(int, infile.readline().strip().split())

    count = 0

    # Tìm tất cả các ước số của k
    for i in range(1, int(k**0.5) + 1):
        if k % i == 0:  # i là ước của k
            j = k // i  # Tính j

            # Kiểm tra (i, j) có hợp lệ không
            if i <= n and j <= n:
                count += 1  # Đếm (i, j)
            if i != j and j <= n and i <= n:
                count += 1  # Đếm (j, i) nếu khác nhau

    with open(output_file, 'w') as outfile:
        outfile.write(f"{count}\n")

count_occurrences(input_path, output_path)
